_posts/2022-06-08-#22778.md

```diff
@@ -0,0 +1,123 @@
+---
+layout: pr
+date: 2022-06-08
+title: "net processing: Reduce resource usage for inbound block-relay-only connections"
+pr: 22778
+authors: [jnewbery]
+components: ["p2p"]
+host: larryruane
+status: upcoming
+commit: 9db82f1
+---
+
+## Notes
+
+- Each of our node's peers is either inbound (our
+  peer initiated the connection) or outbound (we initiated the connection).
+  Once the connection is established, the relationship is symmetric (the
+  protocol works the same in both directions).
+
+- When we initiate an outbound connection, we may decide that we won't
+  relay (forward) transactions or addresses to this peer.
+  This peer is designated a `block-relay-only` peer.
+  This kind of peer connection was covered in earlier review club meetings
+  (PRs [15759](https://bitcoincore.reviews/15759),
+  [19858](https://bitcoincore.reviews/19858),
+  [20726](https://bitcoincore.reviews/20726)).
+  The peer
+  doesn't know that we're treating it as `block-relay-only`.
+
+- When connecting with a peer, we allocate a
+  [`Peer`](https://github.com/bitcoin/bitcoin/blob/1b2e1d179c5b350cac69ee670e01355314f25e11/src/net_processing.cpp#L203)
+  object.
```

Comment:
Maybe add that we allocate a `CNode` object in net (connection layer), and a `Peer` object in net_processing (application layer).